src/components/common/Chart/Chart.tsx

```diff
@@ -96,78 +96,90 @@ function ChartContent(props: ChartContentProps): JSX.Element {
   const theme = useTheme();
 
-  const barThickness = barWidth === undefined ? 50 : barWidth === 0 ? 'flex' : barWidth;
+  const barThickness: number | 'flex' | undefined = barWidth === undefined ? 50 : barWidth === 0 ? 'flex' : barWidth;
 
   useEffect(() => {
     const createChart = async () => {
-      if (chartRef.current) {
-        chartRef.current.destroy();
-        chartRef.current = null;
+      if (chart) {
+        chart.destroy();
       }
 
       const ctx = canvasRef?.current?.getContext('2d');
       if (ctx) {
-        const colors = elementColor ?? generateTerrawareRandomColors(theme, chartData?.labels?.length || 0);
-        chartRef.current = await newChart(locale, ctx, {
-          type,
-          data: {
-            labels: chartData.labels,
-            datasets: chartData.datasets.map((ds, index) => ({
-              label: ds.label,
-              data: ds.values,
-              barThickness,
-              backgroundColor: ds.color ?? colors,
-              minBarLength: 3,
-            })),
-          },
-          options: {
-            maintainAspectRatio: false,
-            layout: {
-              padding: {
-                left: 0,
-                right: 0,
-                top: 10,
-              },
+        setChart(
+          await newChart(locale, ctx, {
+            type,
+            data: {
+              labels: [],
+              datasets: [],
             },
-            plugins: {
-              annotation: barAnnotations,
-              legend: {
-                display: !!showLegend,
-              },
-              tooltip: {
-                displayColors: false,
-                callbacks: {
-                  title: customTooltipTitles
-                    ? ([context]) => {
-                        return customTooltipTitles[context.dataIndex];
-                      }
-                    : undefined,
+            options: {
+              maintainAspectRatio: false,
+              layout: {
+                padding: {
+                  left: 0,
+                  right: 0,
+                  top: 10,
                 },
               },
-            },
-            scales: {
-              y: {
-                ticks: {
-                  precision: 0,
+              scales: {
+                y: {
+                  ticks: {
+                    precision: 0,
+                  },
+                  min: yLimits?.min,
+                  max: yLimits?.max,
                 },
-                min: yLimits?.min,
-                max: yLimits?.max,
               },
             },
-          },
-        });
+          })
+        );
         // when component unmounts
         return () => {
-          chartRef.current?.destroy();
+          chart?.destroy();
         };
       }
     };
     createChart();
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [chartData, locale]);
+  }, [locale]);
+
+  useEffect(() => {
```

Comment:
nice debug/resolution, I don't know much here will let @constanzauanini take a look as well

Reply:
I remember destroying the chart when data changed because if not there was an error that more than 1 chart was referencing the same canvas, but I think I was not able to only update it without creating a new one, so this new approach looks correct to me! Not destroying the chart, but just refreshing its data! Thanks @charlie-tf !